### PR TITLE
finished user form and schema creator

### PR DESCRIPTION
**Pull request by rachelsison**

### CODE REVIEW

File: client/js/components/create_form.jsx [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        AppStore.emitChange();
+      },
+      error: function(error) {
+        console.error(error);
+      }
+    });
+  },
+  createNewQuestion: function(form){
+    console.log("in createNewQuestion2");
+    event.preventDefault();
```
Comment (dduugg) [inline client/js/components/create_form.jsx]: where is `event` defined?

File: client/js/components/create_form.jsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    AppStore.currentQuestionType=[this.refs.questionType.getDOMNode().value];
+    this.trigger();
+  },
+  changeFormTitle: function(){
+    AppStore.formTitle = this.refs.formTitle.getDOMNode().value;
+    this.trigger();
+  },
+  outputUpdate: function(id) {
+    console.log("in outputUpdate function", "#scale"+id);
+            document.querySelector("#output"+id).value = this.refs["scale"+id].getDOMNode().value;
```
Comment (dduugg) [inline client/js/components/create_form.jsx]: indentation is a little weird here

File: client/js/components/create_form.jsx [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  },
+  drop: function(questionIndex) {
+    event.preventDefault();
```
Comment (dduugg) [inline client/js/components/create_form.jsx]: again, not sure where `event` was introduced

File: client/js/components/create_form.jsx [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    console.log("questionIndex in DROP", questionIndex);
+    var targetIndex = AppStore.questionDropTarget = questionIndex; 
+    var movingIndex = AppStore.questionBeingDragged;    
+    this.toggleQuestionActive(questionIndex);    
+    AppStore.questionsInEdit[AppStore.questionBeingDragged][0]=questionIndex;         
+    AppStore.questionsInEdit[targetIndex][0]=movingIndex;
+    var swap = AppStore.questionsInEdit[movingIndex];
```
Comment (dduugg) [inline client/js/components/create_form.jsx]: define all variables at the [top](http://javascript.crockford.com/code.html) of the function

File: client/js/components/create_form.jsx [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+                          onDragEnter={this.toggleQuestionActive.bind(this, questionIndex)} 
+                          onDragLeave={this.toggleQuestionActive.bind(this, questionIndex)} 
+                          onDragOver={this.preventDefault}
+                          onDrop={this.drop.bind(this, questionIndex)}>
+                    <div id="qTitle">{questionTitle}</div>
+                        <input type="text" value="answer goes here" readOnly="true"/>
+                    </div>;
+                  
+                  
+              case "scale":
```
Comment (dduugg) [inline client/js/components/create_form.jsx]: do you mean to fall through? or does react do something magical?

File: client/js/components/create_form.jsx [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+      <button onClick={this.createFormSchema} className="completeForm">Publish Form</button>
+      </div>
+    </div>
+  </div>
+      );}
+
+
+});
+
+module.exports = CreateForm;
```
Comment (dduugg) [inline client/js/components/create_form.jsx]: newline

File: client/js/components/create_form.jsx [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+    AppStore.questionsInEdit[movingIndex]=AppStore.questionsInEdit[targetIndex];
+    AppStore.questionsInEdit[targetIndex] = swap;
+    this.trigger();
+  },
+
+  dragStart: function(questionIndex) {
+    AppStore.questionBeingDragged = questionIndex;
+  },
+
+  preventDefault: function(event) {
```
Comment (zindlerb) [inline client/js/components/create_form.jsx]: Why is this function here?

File: client/js/components/create_form.jsx [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+  componentDidMount: function() {
+    AppStore.addChangeListener(this.forceUpdate);
+  },
+
+  componentWillUnmount: function() {
+    AppStore.removeChangeListener(this.forceUpdate);
+  },
+
+  trigger: function(){
+    this.forceUpdate();
```
Comment (zindlerb) [inline client/js/components/create_form.jsx]: Why do we trigger a force update? We should be passing the data down through props then emitting a change event in the app store.

File: client/js/components/create_form.jsx [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+  },
+
+  createFormSchema: function(event){
```
Comment (zindlerb) [inline client/js/components/create_form.jsx]: This function should be in the app store. And then we could call the router in a callback.

File: client/js/components/create_form.jsx [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+        AppStore.emitChange();
+      },
+      error: function(error) {
+        console.error(error);
+      }
+    });
+  },
+
+  createNewQuestion: function(form){
+    console.log("in createNewQuestion2");
```
Comment (zindlerb) [inline client/js/components/create_form.jsx]: Probs should remove this console.log

**Pull request merged**